symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/MessageService.java

```diff
@@ -54,6 +148,121 @@ public V4Message send(String streamId, String message) {
+   */
+  public V4Message getMessage(String messageId) {
+    try {
+      return this.messagesApi.v1MessageIdGet(authSession.getSessionToken(), authSession.getKeyManagerToken(),
+          messageId);
+    } catch (ApiException e) {
+      log.error("Cannot get message", e);
+      sleep(1_000);
+      try {
+        authSession.refresh();
+      } catch (AuthUnauthorizedException exception) {
+        log.error("Cannot authenticate", exception);
+        return null;
+      }
+      return this.getMessage(messageId);
+    }
+  }
+
+  /**
+   * Imports a list of messages to Symphony
+   *
+   * @param messages the list of messages to be imported
+   * @return the list of imported messages
+   * @see <a href="https://developers.symphony.com/restapi/reference#import-message-v4">Import Message</a>
+   */
+  public List<V4ImportResponse> importMessages(List<V4ImportedMessage> messages) {
+    try {
+      return this.messagesApi.v4MessageImportPost(authSession.getSessionToken(), authSession.getKeyManagerToken(),
+          messages);
+    } catch (ApiException e) {
+      log.error("Cannot import messages", e);
+      sleep(1_000);
+      try {
+        authSession.refresh();
+      } catch (AuthUnauthorizedException exception) {
+        log.error("Cannot authenticate", exception);
+        return null;
+      }
+      return this.importMessages(messages);
+    }
+  }
+
+  private void fetchTemplate(String template) throws TemplateException {
+    if (isBuiltInTemplate(template)) {
+      tryFetchBuiltInTemplate(template);
+    }
+
+    if (this.template == null && isValidUrlSyntax(template)) {
+      tryFetchTemplateFromUrl(template);
+    }
+
+    if (this.template == null) {
+      tryFetchTemplateFromClasspath(template);
+    }
+
+    if (this.template == null) {
+      tryFetchTemplateFromFileSystem(template);
+    }
+
+    if (this.template == null) {
+      throw new TemplateException("Template " + template + " not found", null);
+    }
+  }
+
+  private boolean isBuiltInTemplate(String template) {
+    return this.templateEngine.getBuiltInTemplates().contains(template);
+  }
+
+  private void tryFetchBuiltInTemplate(String template) {
+    try {
+      this.template = this.templateEngine.newBuiltInTemplate(template);
+    } catch (TemplateException e) {
+      log.debug("{} is not a built-in template", template);
+    }
+  }
+
+  private void tryFetchTemplateFromClasspath(String template) {
+    try {
+      this.template = this.templateEngine.newTemplateFromClasspath(template);
+    } catch (TemplateException e) {
+      log.debug("{} is not found in classpath", template);
+    }
+  }
+
+  private void tryFetchTemplateFromFileSystem(String template) {
+    try {
+      this.template = this.templateEngine.newTemplateFromFile(template);
+    } catch (TemplateException e) {
+      log.debug("{} is not found on file system", template);
+    }
+  }
+
+  private boolean isValidUrlSyntax(String templateUrl) {
```

Comment:
take care of that, if we have other components applying the same validation we should create a utility class for!!!